pkg/node/sendoperation/callsc/callsc.go

```diff
@@ -89,3 +105,84 @@ func (c *CallSC) Message() []byte {
 
 	return msg
 }
+
+//nolint:funlen, cyclop
+func DecodeMessage(data []byte) (*MessageContent, error) {
+	callSCContent := &MessageContent{}
+	buf := bytes.NewReader(data)
+
+	// Read operationId
+	callSCOpID, err := binary.ReadUvarint(buf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CallSCOpID: %w", err)
+	}
+
+	callSCContent.OperationID = callSCOpID
+
+	// Read maxGas
+	gasLimit, err := binary.ReadUvarint(buf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read gasLimit: %w", err)
+	}
+
+	callSCContent.GasLimit = gasLimit
+
+	// Read Coins
+	coins, err := binary.ReadUvarint(buf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read coins: %w", err)
+	}
+
+	callSCContent.Coins = coins
+
+	// Read target address length
+	addressLength, err := binary.ReadUvarint(buf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read address length: %w", err)
+	}
+
+	address := make([]byte, addressLength)
+
+	_, err = buf.Read(address)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read address: %w", err)
+	}
```

Comment:
```suggestion
	assert.NoError(t, err)
```

Reply:
why assert , this is [callsc.go], why we should add the assert ?

Reply:
you're right, I would like to show an example of the assert package, related to https://github.com/massalabs/station/pull/1082#discussion_r1288536795 but I should have done this in a unit test file